backend/layers/processing/process_download.py

```diff
@@ -75,37 +91,106 @@ def remove_prefix(string: str, prefix: str) -> str:
         else:
             return string
 
-    def process(self, dataset_id: DatasetVersionId, dataset_uri: str, artifact_bucket: str, sfn_task_token: str):
+    @staticmethod
+    def get_job_definion_name(dataset_vid: str) -> str:
+        if os.getenv("REMOTE_DEV_PREFIX"):
+            stack_name = os.environ["REMOTE_DEV_PREFIX"].replace("/", "")
+            prefix = f"{os.environ['DEPLOYMENT_STAGE']}-{stack_name}"
+        else:
+            prefix = f"{os.environ['DEPLOYMENT_STAGE']}"
+        job_definition_name = f"dp-{prefix}-ingest-process-{dataset_vid}"
+        return job_definition_name
+
+    @staticmethod
+    def estimate_resource_requirements(
+        adata: scanpy.AnnData,
+        memory_modifier: float = MEMORY_MODIFIER,
+        min_memory_MB: int = MIN_MEMORY_MB,
+        min_vcpu: int = MIN_VCPU,
+        max_memory_MB: int = MAX_MEMORY_MB,
+        max_vcpu: int = MAX_VCPU,
+        swap_memory_MB: int = SWAP_MEMORY_MB,
+        memory_per_vcpu: int = MEMORY_PER_VCPU,
+    ) -> Dict[str, Any]:
```

Comment:
Looks like the values are all integers here.

Reply:
```suggestion
    ) -> Dict[str, int]:
```